app/src/main/kotlin/ExtendedStatelessSolver.kt

```diff
@@ -0,0 +1,40 @@
+import kotlin.math.*
+
+class ExtendedStatelessSolver<TState, TAction>(
+        mdp: MDP<TState, TAction>,
+        simulationDepthLimit: Int,
+        explorationConstant: Double,
+        rewardDiscountFactor: Double,
+        verbose: Boolean
+) : StatelessSolver<TState, TAction>(mdp, simulationDepthLimit, explorationConstant, rewardDiscountFactor, verbose) {
+
+    val rewardHistory = mutableListOf<Double>()
+
+    override fun constructTree(iterations: Int) {
```

Comment:
override this to get rewards.

Reply:
You can't override a function's return value. Additional functionalities should be implemented as new functions/properties.

Reply:
debugging purposes